### PR TITLE
Fix / route with prefix in fastify@v3.10.1

### LINKED ISSUE #97: Fastify 3.10 breaks prefixed plugins
## 🐛 Bug Report

Fastify 3.10, together with the use of prefixes, breaks `fastify-websocket`.

```ts
fastify.register(
  (fastify, opts, done) => {
    fastify.get("/", { websocket: true }, (connection, req) => {
      connection.socket.send("This works");
    });

    done();
  },
  { prefix: "/activity" }
);
```

## To Reproduce

- Clone https://github.com/Skn0tt/fastify-websocket-3.10-repro
- Run `index.js`
- Connect to `ws://localhost:3000/activity`

## Your Environment

- *node version*: 15.3
- *fastify version*: >=3.10
- *os*: Mac

## Some ideas as to what may be causing this

Prior to 3.10, this line was called twice per route:

https://github.com/fastify/fastify-websocket/blob/99c94a003da2353c1f482301027166e64e743582/index.js#L36

That's why this lines makes sure to ignore one of them:

https://github.com/fastify/fastify-websocket/blob/99c94a003da2353c1f482301027166e64e743582/index.js#L42

Now with 3.10, the `onRoute` is called only once, and _all_ of the routes are effectively ignored.

> originally encountered in https://github.com/quirrel-dev/quirrel/issues/49

### PR DESCRIPTION
**Pull request by mcollina**
Fix #97 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by kibertoad — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -113,6 +110,6 @@ function close (fastify, done) {
 }
 
 module.exports = fp(fastifyWebsocket, {
-  fastify: '3.x',
+  fastify: '>= 3.10.1',
```
Comment (kibertoad) [inline index.js:113]: This does not qualify as a breaking change?

**Review by Skn0tt — APPROVED** [review]
If this really is the fix - then why was it necessary before? Was the Double-onRoute a bug in Fastify? Because otherwise, that‘d be a breaking change in 3.10 (a Minor) that‘s mentioned nowhere in the release notes

**Comment by mcollina** [comment]
This was caused by: https://github.com/fastify/fastify/pull/2782/files. We had a few bugs in other plugins because it was called twice.

(Note that the change in Fastify was _not_ causing any tests failures here).

cc @salmanm

**Review by delvedor — APPROVED** [review]
LGTM

We should add a similar test in Fastify core.

File: index.js (hunk 1 above)
Comment (delvedor) [reply index.js:113]: Nope, in a real world application you should use lock files, so this wouldn't impact you until you want to upgrade.

**Comment by salmanm** [comment]
@Skn0tt double `onRoute` was called when `prefixTrailingSlash` was set to `both`. However, this behavior was neither expected (or at least I think so, coz single route definition triggering two `onRoute`) nor documented. Hence, we removed this double trigger.

**Pull request merged**